merlin-worker/src/main/java/gov/nasa/jpl/aerie/merlin/worker/MerlinWorkerAppDriver.java

```diff
@@ -84,7 +86,9 @@ public static void main(String[] args) throws InterruptedException {
             notification.simulationTemplateRevision());
         final ResultsProtocol.WriterRole writer = owner.get();
         try {
-          simulationAgent.simulate(planId, revisionData, writer);
+          canceledListener.register(new DatasetId(datasetId));
```

Comment:
 🔶 Some food for thought - I think moving `register` to before `claim` avoids a possible race condition where a cancel request comes in after `claim`, but before `register`

```diff
final var datasetId = notification.datasetId();
+
+canceledListener.register(new DatasetId(datasetId));
final Optional<ResultsProtocol.OwnerRole> owner = stores.results().claim(planId, datasetId);
if (owner.isEmpty()) continue;

final var revisionData = new PostgresPlanRevisionData(
    notification.modelRevision(),
    notification.planRevision(),
    notification.simulationRevision(),
    notification.simulationTemplateRevision());
final ResultsProtocol.WriterRole writer = owner.get();
try {
-  canceledListener.register(new DatasetId(datasetId));
  simulationAgent.simulate(planId, revisionData, writer, canceledListener);
```

Reply:
Ah, thanks for the catch! Done